harness/determined/deploy/cli.py

```diff
@@ -1,40 +1,50 @@
 import argparse
-import sys
+import warnings
+from typing import List, Union
 
-import determined
-import determined.deploy.aws.cli
-import determined.deploy.gcp.cli
-import determined.deploy.local.cli
+from determined import __version__
+from determined.cli.declarative_argparse import Arg, Cmd, add_args
 
+from .aws.cli import args_description as aws_args_description
+from .gcp.cli import args_description as gcp_args_description
+from .local.cli import args_description as local_args_description
 
-def main() -> None:
-    environment_map = {
-        "aws": determined.deploy.aws.cli.deploy_aws,
-        "gcp": determined.deploy.gcp.cli.deploy_gcp,
-        "local": determined.deploy.local.cli.deploy_local,
-    }
+args_subs: List[Union[Arg, Cmd]] = [
+    # TODO(ilia): Remove --version flag when det-deploy is deprecated.
```

Comment:
can you create a ticket for that and put the ticket number here rather than your name?

Reply:
done